Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -583,4 +585,116 @@ final class DocumentTests: MongoSwiftTestCase {
+
+            // doc.formIndex
+            var firstIndex = 0
+            doc.formIndex(after: &firstIndex)
+            expect(firstIndex).to(equal(doc.index(after: doc.startIndex)))
+
+            // doc.index(offsetBy:), doc.index(offsetBy:,limitedBy:)
+            expect(doc.index(doc.startIndex, offsetBy: 2)).to(equal(doc.endIndex))
+            expect(doc.index(doc.startIndex, offsetBy: 2, limitedBy: doc.endIndex)).to(equal(doc.endIndex))
+            expect(doc.index(doc.startIndex, offsetBy: 99, limitedBy: 1)).to(beNil())
+
+            // firstIndex(where:); This line is commented out because Travis currently builds on 9.4, but this needs 10+
+//            expect(doc.firstIndex { $0.key == "a" && bsonEquals($0.value, 3) }).to(equal(doc.startIndex))
+        }
+
+        fileprivate static func testMutators() throws {
+            var doc: Document = ["a": 3, "b": 2, "c": 5, "d": 4]
+
+            // doc.removeFirst
+            let firstElem = doc.removeFirst()
+            expect(firstElem.key).to(equal("a"))
+            expect(firstElem.value).to(bsonEqual(3))
+            expect(doc).to(equal(["b": 2, "c": 5, "d": 4]))
+            expect(doc).to(haveCorrectCount())
+
+            // doc.removeFirst(k:)
+            doc.removeFirst(2)
+            expect(doc).to(equal(["d": 4]))
+            expect(doc).to(haveCorrectCount())
+
+            // doc.popFirst
+            let lastElem = doc.popFirst()
+            expect(lastElem?.key).to(equal("d"))
+            expect(lastElem?.value).to(bsonEqual(4))
+            expect(doc).to(equal([:]))
+            expect(doc).to(haveCorrectCount())
+
+            // doc.merge
+            let newDoc: Document = ["e": 4, "f": 2]
+            try doc.merge(newDoc)
+            expect(doc).to(haveCorrectCount())
+        }
+
+        fileprivate static func testPrefixSuffix() {
+            let doc: Document = ["a": 3, "b": 2, "c": 5, "d": 4, "e": 3]
+
+            let upToPrefixDoc = doc.prefix(upTo: 2)
+            let throughPrefixDoc = doc.prefix(through: 1)
+            let suffixDoc = doc.suffix(from: 1)
+
+            // doc.prefix(upTo:)
+            expect(upToPrefixDoc).to(equal(["a": 3, "b": 2]))
+            expect(upToPrefixDoc).to(haveCorrectCount())
+
+            // doc.prefix(through:)
+            expect(throughPrefixDoc).to(equal(["a": 3, "b": 2]))
+            expect(throughPrefixDoc).to(haveCorrectCount())
+
+            // doc.suffix
+            expect(suffixDoc).to(equal(["b": 2, "c": 5, "d": 4, "e": 3]))
+            expect(suffixDoc).to(haveCorrectCount())
+        }
+    }
+
+    func testDocumentConformanceToCollection() throws {
+        CollectionConformance.testDocumentIndexLogic()
+        try CollectionConformance.testMutators()
+        CollectionConformance.testPrefixSuffix()
+    }
+}
+
+/// A Nimble matcher for testing that the count of a Document is what it should be.
```

Comment:
let's clarify that this is specifically checking that the count determined by libbson is equal to our own bookkeeping number.

Reply:
Done.